pkg/services/cloudmigration/cloudmigrationimpl/cloudmigration.go

```diff
@@ -464,8 +464,24 @@ func (s *Service) GetMigrationRunList(ctx context.Context, migUID string) (*clou
 	return runList, nil
 }
 
-func (s *Service) DeleteSession(ctx context.Context, uid string) (*cloudmigration.CloudMigrationSession, error) {
-	c, err := s.store.DeleteMigrationSessionByUID(ctx, uid)
+func (s *Service) DeleteSession(ctx context.Context, sessionUID string) (*cloudmigration.CloudMigrationSession, error) {
+	// first we try to delete all the associated information to the session
+	snapshots, err := s.store.GetMigrationStatusList(ctx, sessionUID)
```

Comment:
Can we use a transaction to avoid deleting only part of what needs to be deleted?

Reply:
done! I moved them to the sql store and run everything within a transaction